File: examples/companion-packages/pub-app/lib/src/flags.dart
/// TinyFlags definitions for PawMatch.
library;

import 'package:agentxm_example_tinyflags/agentxm_example_tinyflags.dart';

const String homeCheckFollowup = 'home-check-followup';
const String feeBreakdownDetailed = 'fee-breakdown-detailed';
const String longStayHighlight = 'long-stay-highlight';
const String suggestDonateAfterAdoption = 'suggest-donate-after-adoption';
const String showCharityRatings = 'show-charity-ratings';
const String recommendationStrategy = 'recommendation-strategy';
const String matchQuizDepth = 'match-quiz-depth';
const String petCardStyle = 'pet-card-style';
const String donateFocusDefault = 'donate-focus-default';

/// Build the TinyFlags client with PawMatch's flag definitions.
TinyFlags createFlags() {
  return TinyFlags({
    homeCheckFollowup: BooleanFlag(defaultValue: false, rollout: 25),
    feeBreakdownDetailed: BooleanFlag(defaultValue: true),
    longStayHighlight: BooleanFlag(defaultValue: true),
    suggestDonateAfterAdoption: BooleanFlag(defaultValue: false, rollout: 50),
    showCharityRatings: BooleanFlag(defaultValue: true),
    recommendationStrategy: VariantFlag(
      variants: ['popularity', 'match-quiz', 'longest-stay'],
      defaultValue: 'match-quiz',
      rollout: {'longest-stay': 20},
    ),
    matchQuizDepth: VariantFlag(
      variants: ['short', 'standard', 'thorough'],
      defaultValue: 'standard',
    ),
    petCardStyle: VariantFlag(
      variants: ['compact', 'detailed', 'playful'],
      defaultValue: 'detailed',
    ),
    donateFocusDefault: VariantFlag(
      variants: ['all', 'shelters', 'rescue'],
      defaultValue: 'all',
    ),
  });
}

File: examples/companion-packages/pypi-python-app/src/agentxm_example_pawmatch/flags.py
"""TinyFlags definitions for PawMatch."""

from agentxm_example_tinyflags import BooleanFlag, Flag, TinyFlags, VariantFlag

HOME_CHECK_FOLLOWUP = "home-check-followup"
FEE_BREAKDOWN_DETAILED = "fee-breakdown-detailed"
LONG_STAY_HIGHLIGHT = "long-stay-highlight"
SUGGEST_DONATE_AFTER_ADOPTION = "suggest-donate-after-adoption"
SHOW_CHARITY_RATINGS = "show-charity-ratings"
RECOMMENDATION_STRATEGY = "recommendation-strategy"
MATCH_QUIZ_DEPTH = "match-quiz-depth"
PET_CARD_STYLE = "pet-card-style"
DONATE_FOCUS_DEFAULT = "donate-focus-default"


def create_flags() -> TinyFlags:
    """Build the TinyFlags client with PawMatch's flag definitions."""
    definitions: dict[str, Flag] = {
        HOME_CHECK_FOLLOWUP: BooleanFlag(default=False, rollout=25),
        FEE_BREAKDOWN_DETAILED: BooleanFlag(default=True),
        LONG_STAY_HIGHLIGHT: BooleanFlag(default=True),
        SUGGEST_DONATE_AFTER_ADOPTION: BooleanFlag(default=False, rollout=50),
        SHOW_CHARITY_RATINGS: BooleanFlag(default=True),
        RECOMMENDATION_STRATEGY: VariantFlag(
            variants=("popularity", "match-quiz", "longest-stay"),
            default="match-quiz",
            rollout={"longest-stay": 20},
        ),
        MATCH_QUIZ_DEPTH: VariantFlag(
            variants=("short", "standard", "thorough"),
            default="standard",
        ),
        PET_CARD_STYLE: VariantFlag(
            variants=("compact", "detailed", "playful"),
            default="detailed",
        ),
        DONATE_FOCUS_DEFAULT: VariantFlag(
            variants=("all", "shelters", "rescue"),
            default="all",
        ),
    }
    return TinyFlags(definitions)
